### PR TITLE
chore: add docs and helper for testing samples locally

### PR DESCRIPTION
**Pull request by bharathkkb**
This adds instructions and some makefile updates to enable local testing.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Set the following environment variables.
+```
+export TF_VAR_org_id="your_org_id"
+export TF_VAR_folder_id="your_folder_id"
+export TF_VAR_billing_account="your_billing_account_id"
+```
+
+With these settings in place, you can prepare a test project using Docker:
+
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:36]: Add step to run:

gcloud auth application-default login

**Comment by betsy-lichtenberg** [comment]
I'm getting these errors when I run `make docker_test_prepare`

│ Error: Invalid provider configuration
│ 
│ Provider "registry.terraform.io/hashicorp/google-beta" requires explicit configuration. Add a provider block to the root module
│ and configure the provider's required arguments as described in the provider documentation.
│ 
╵
╷
│ Error: Invalid provider configuration
│ 
│ Provider "registry.terraform.io/hashicorp/google" requires explicit configuration. Add a provider block to the root module and
│ configure the provider's required arguments as described in the provider documentation.
│ 
╵
╷
│ Error: Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded. To use your gcloud credentials, run 'gcloud auth application-default login'.  Original error: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
│ 
│   with provider["registry.terraform.io/hashicorp/google-beta"],
│   on <empty> line 0:
│   (source code not available)
│ 
╵
╷
│ Error: Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded. To use your gcloud credentials, run 'gcloud auth application-default login'.  Original error: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
│ 
│   with provider["registry.terraform.io/hashicorp/google"],
│   on <empty> line 0:
│   (source code not available)
│ 
╵
make: *** [docker_test_prepare] Error 1

**Comment by bharathkkb** [comment]
@betsy-lichtenberg Did the error go away after running `gcloud auth application-default login`?

**Comment by betsy-lichtenberg** [comment]
No

**Comment by bharathkkb** [comment]
@betsy-lichtenberg Are you testing from this branch? The logs you pasted above showed
```
make docker_test_prepare
docker run --rm -it \
		-e SERVICE_ACCOUNT_JSON \
```
I have made an update to that makefile target in this branch, so it is now using your local credentials.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./CONTRIBUTING.md:15:Integration tests are used to verify that the samples are actuatable 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ```

**Comment by betsy-lichtenberg** [comment]
That helped! Now I'm getting these errors when I run `make docker_test_prepare`:


```
│ Error: failed pre-requisites: missing permission on "billingAccounts/01B783-CE7C44-CAFEF8": billing.resourceAssociations.create
│ 
│   with module.projects[3].module.project-factory.google_project.main,
│   on .terraform/modules/projects/modules/core_project_factory/main.tf line 65, in resource "google_project" "main":
│   65: resource "google_project" "main" {
│ 
╵
╷
│ Error: failed pre-requisites: missing permission on "billingAccounts/01B783-CE7C44-CAFEF8": billing.resourceAssociations.create
│ 
│   with module.projects[2].module.project-factory.google_project.main,
│   on .terraform/modules/projects/modules/core_project_factory/main.tf line 65, in resource "google_project" "main":
│   65: resource "google_project" "main" {
│ 
╵
╷
│ Error: failed pre-requisites: missing permission on "billingAccounts/01B783-CE7C44-CAFEF8": billing.resourceAssociations.create
│ 
│   with module.projects[0].module.project-factory.google_project.main,
│   on .terraform/modules/projects/modules/core_project_factory/main.tf line 65, in resource "google_project" "main":
│   65: resource "google_project" "main" {
│ 
╵
╷
│ Error: failed pre-requisites: missing permission on "billingAccounts/01B783-CE7C44-CAFEF8": billing.resourceAssociations.create
│ 
│   with module.projects[1].module.project-factory.google_project.main,
│   on .terraform/modules/projects/modules/core_project_factory/main.tf line 65, in resource "google_project" "main":
│   65: resource "google_project" "main" {
│ 
╵
make: *** [docker_test_prepare] Error 1
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+The easiest way to test the samples are in isolated test projects. The
+setup for projects are defined in [test/setup](./test/setup/)
+directory.
+
+To use this setup, you need to authenticate gcloud with an identity that has
+Project Creator access on a folder and the Billing Account User role on a billing account.
+
+Set the following environment variables.
+```
+export TF_VAR_org_id="your_org_id"
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:30]: I set these to the project info that I typically use for testing:

```
 export TF_VAR_org_id="compute-writers.joonix.net"
 export TF_VAR_folder_id="individual-projects"
 export TF_VAR_billing_account="01B783-CE7C44-CAFEF8"
```

Should I be setting them to something else?

**Comment by bharathkkb** [comment]
This is likely because your user account does not have billing user on `01B783-CE7C44-CAFEF8`. Could you try granting it using https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/docs/TROUBLESHOOTING.md#missing-rolesbillinguser-role

File: CONTRIBUTING.md (hunk 2 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:30]: The `TF_VAR_folder_id` should be just the numeric id. Other than that, it looks good. The other issue maybe permissions as I mentioned below.

Comment (betsy-lichtenberg) [reply CONTRIBUTING.md:30]: Yeah it looks like I'm not a billing admin. I'm trying to get that fixed, but in the meantime maybe another reviewer can test the procedure so I don't hole it up.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+export TF_VAR_folder_id="your_folder_id"
+export TF_VAR_billing_account="your_billing_account_id"
+```
+
+With these settings in place, you can prepare a test project using Docker:
+
+```
+cd terraform-docs-samples
+make docker_test_prepare
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:39]: For me, this runs successfully in Cloud Shell.

On my Mac, I get the following error:

```
$ make docker_test_prepare
docker run --rm -it \
		-e SERVICE_ACCOUNT_JSON \
		-e TF_VAR_org_id \
		-e TF_VAR_folder_id \
		-e TF_VAR_billing_account \
		-v "/Users/betsyl/terraform/terraform-docs-samples":/workspace \
		gcr.io/cloud-foundation-cicd/cft/developer-tools:1 \
		/usr/local/bin/execute_with_credentials.sh prepare_environment
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
ERROR: (gcloud.auth.activate-service-account) Missing required argument [ACCOUNT]: An account is required when using .p12 keys
ERROR: (gcloud.auth.activate-service-account) Missing required argument [ACCOUNT]: An account is required when using .p12 keys
make: *** [docker_test_prepare] Error 1
```

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./CONTRIBUTING.md:15:Integration tests are used to verify that the samples are actuatable 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate ./cloud_run_service_add_tag
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_configuration
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_connect_cloud_sql
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_custom_domain_mapping
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_deploy_tag
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_image_processing
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_ingress
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_interservice
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_multiple_regions
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_noauth
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_pubsub
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_remove_tag
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_scheduled
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_secure_services
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_static_outbound
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_tasks
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_traffic_gradual_rollout
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_traffic_latest_revision
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_traffic_rollback
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_traffic_split
Success!
 The configuration is valid.
terraform_validate ./cloud_run_service_traffic_split_tag
Success!
 The configuration is valid.
terraform_validate ./cloud_run_system_packages
Success!
 The configuration is valid.
terraform_validate ./cloudfunctions2_basic
Success!
 The configuration is valid.
terraform_validate ./cloudfunctions2_basic_auditlogs
Success!
 The configuration is valid.
terraform_validate ./cloudfunctions2_basic_gcs
Success!
 The configuration is valid.
terraform_validate ./cloudfunctions2_full
Success!
 The configuration is valid.
terraform_validate ./cloudrun_service_access_control
Success!
 The configuration is valid.
terraform_validate ./cloudrun_service_identity
Success!
 The configuration is valid.
terraform_validate ./cloudrun_vpc_access_connector
Success!
 The configuration is valid.
terraform_validate ./composer_multi_project_monitoring
Success!
 The configuration is valid.
terraform_validate ./compute_packet_mirroring_full
Success!
 The configuration is valid.
terraform_validate ./compute_reservation
Success!
 The configuration is valid.
terraform_validate ./compute_run_shutdown_scripts
Success!
 The configuration is valid.
terraform_validate ./dns_managed_zone_basic
Success!
 The configuration is valid.
terraform_validate ./dns_managed_zone_private
Success!
 The configuration is valid.
terraform_validate ./dns_managed_zone_private_forwarding
Success!
 The configuration is valid.
terraform_validate ./dns_managed_zone_private_peering
Success!
 The configuration is valid.
terraform_validate ./dns_managed_zone_quickstart
Success!
 The configuration is valid.
terraform_validate ./dns_managed_zone_service_directory
Success!
 The configuration is valid.
terraform_validate ./dns_policy_basic
Success!
 The configuration is valid.
terraform_validate ./dns_record_set_basic
Success!
 The configuration is valid.
terraform_validate ./dns_response_policy_basic
Success!
 The configuration is valid.
terraform_validate ./dns_response_policy_rule_basic
Success!
 The configuration is valid.
terraform_validate ./eventarc_basic_tf
Success!
 The configuration is valid.
terraform_validate ./eventarc_workflows
Success!
 The configuration is valid.
terraform_validate ./external_cdn_lb_with_backend_bucket
Success!
 The configuration is valid.
terraform_validate ./external_http_lb_bucket_backend
Success!
 The configuration is valid.
terraform_validate ./external_http_lb_mig_backend
Success!
 The configuration is valid.
terraform_validate ./external_http_lb_mig_backend_custom_header
Success!
 The configuration is valid.
terraform_validate ./external_ssl_proxy_lb_mig_backend
Success!
 The configuration is valid.
terraform_validate ./external_ssl_proxy_lb_mig_backend_custom_header
Success!
 The configuration is valid.
terraform_validate ./external_tcp_proxy_lb_mig_backend
Success!
 The configuration is valid.
terraform_validate ./external_tcp_proxy_lb_mig_backend_custom_header
Success!
 The configuration is valid.
terraform_validate ./external_vpn_gateway
Success!
 The configuration is valid.
terraform_validate ./firewall_with_target_tags
Success!
 The configuration is valid.
terraform_validate ./flask_google_cloud_quickstart
Success!
 The configuration is valid.
terraform_validate ./ha_vpn_gateway_gcp_to_gcp
Success!
 The configuration is valid.
terraform_validate ./health-check-tcp
Success!
 The configuration is valid.
terraform_validate ./instance_custom_hostname
Success!
 The configuration is valid.
terraform_validate ./instance_virtual_display_enabled
Success!
 The configuration is valid.
terraform_validate ./int_https_lb_https_redirect
Success!
 The configuration is valid.
terraform_validate ./internal_http_lb_with_mig_backend
Success!
 The configuration is valid.
terraform_validate ./internal_tcp_udp_lb_with_mig_backend
Success!
 The configuration is valid.
terraform_validate ./media_cdn_keyset
Success!
 The configuration is valid.
terraform_validate ./media_cdn_quickstart
Success!
 The configuration is valid.
terraform_validate ./network_custom_mtu
Success!
 The configuration is valid.
terraform_validate ./network_management_connectivity_test_addresses
Success!
 The configuration is valid.
terraform_validate ./network_management_connectivity_test_instances
Success!
 The configuration is valid.
terraform_validate ./os_login
Success!
 The configuration is valid.
terraform_validate ./private_service_connect_google_apis
Success!
 The configuration is valid.
terraform_validate ./privateca_capool_all_fields
Success!
 The configuration is valid.
terraform_validate ./privateca_capool_basic
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_authority_basic
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_authority_byo_key
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_authority_subordinate
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_config
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_csr
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_no_authority
Success!
 The configuration is valid.
terraform_validate ./privateca_certificate_with_template
Success!
 The configuration is valid.
terraform_validate ./privateca_quickstart
Success!
 The configuration is valid.
terraform_validate ./region_autoscaler_basic
Success!
 The configuration is valid.
terraform_validate ./regional-external-http-load-balancer
Success!
 The configuration is valid.
terraform_validate ./regional_external_http_load_balancer
Success!
 The configuration is valid.
terraform_validate ./shared_vpc_http_ilb_with_mig_backend
Success!
 The configuration is valid.
terraform_validate ./shared_vpc_http_ilb_with_mig_backend_basic_example
Success!
 The configuration is valid.
terraform_validate ./shared_vpc_http_ilb_with_mig_backend_cross_reference_example
Success!
 The configuration is valid.
terraform_validate ./shared_vpc_https_ilb_with_mig_backend_basic_example
Success!
 The configuration is valid.
terraform_validate ./shared_vpc_https_ilb_with_mig_backend_cross_reference_example
Success!
 The configuration is valid.
terraform_validate ./spot_instance_basic
Success!
 The configuration is valid.
terraform_validate ./sql_database_basic
Success!
 The configuration is valid.
terraform_validate ./sql_database_instance_my_sql
Success!
 The configuration is valid.
terraform_validate ./sql_database_instance_postgres
Success!
 The configuration is valid.
terraform_validate ./sql_database_instance_sqlserver
Success!
 The configuration is valid.
terraform_validate ./sql_instance_cmek
Success!
 The configuration is valid.
terraform_validate ./sql_instance_ha
Success!
 The configuration is valid.
terraform_validate ./sql_instance_iam_condition
Success!
 The configuration is valid.
terraform_validate ./sql_instance_labels
Success!
 The configuration is valid.
terraform_validate ./sql_instance_pitr
Success!
 The configuration is valid.
terraform_validate ./sql_instance_ssl_cert
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_authorized_network
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_backup
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_backup_location
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_backup_retention
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_clone
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_flags
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_private_ip
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_public_ip
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_pvp
╷
│
 
Error: 
Failed to query available provider packages
│
 
│
 
Could not retrieve the list of available versions for provider
│
 
hashicorp/google: could not connect to registry.terraform.io: Failed to
│
 
request discovery document: Get
│
 
"https://registry.terraform.io/.well-known/terraform.json": write tcp
│
 
192.168.10.2:54862->151.101.2.49:443: write: connection reset by peer
╵
terraform_validate ./sql_mysql_instance_replica
Success!
 The configuration is valid.
terraform_validate ./sql_mysql_instance_ssm
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_authorized_network
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_backup
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_backup_location
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_backup_retention
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_clone
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_flags
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_private_ip
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_public_ip
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_pvp
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_replica
Success!
 The configuration is valid.
terraform_validate ./sql_postgres_instance_ssm
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_active_directory
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_authorized_network
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_backup
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_backup_location
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_backup_retention
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_clone
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_flags
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_private_ip
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_public_ip
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_replica
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_instance_ssm
Success!
 The configuration is valid.
terraform_validate ./sql_sqlserver_vm_instance
Success!
 The configuration is valid.
terraform_validate ./storage_hmac_key
Success!
 The configuration is valid.
terraform_validate ./storage_make_data_public
Success!
 The configuration is valid.
terraform_validate ./storage_new_bucket
Success!
 The configuration is valid.
terraform_validate ./storage_object_lifecycle_setting
Success!
 The configuration is valid.
terraform_validate ./storage_pubsub_notifications
Success!
 The configuration is valid.
terraform_validate ./storage_static_website
Success!
 The configuration is valid.
terraform_validate ./target_grpc_proxy_basic
Success!
 The configuration is valid.
terraform_validate ./target_http_proxy_basic
Success!
 The configuration is valid.
terraform_validate ./target_http_proxy_https_redirect
Success!
 The configuration is valid.
terraform_validate ./target_https_proxy_basic
Success!
 The configuration is valid.
terraform_validate ./target_ssl_proxy_basic
Success!
 The configuration is valid.
terraform_validate ./target_tcp_proxy_basic
Success!
 The configuration is valid.
terraform_validate ./test/setup
Success!
 The configuration is valid.
terraform_validate ./url_map_bucket_and_service
Success!
 The configuration is valid.
terraform_validate ./url_map_header_based_routing
Success!
 The configuration is valid.
terraform_validate ./url_map_parameter_based_routing
Success!
 The configuration is valid.
terraform_validate ./url_map_traffic_director_path
Success!
 The configuration is valid.
terraform_validate ./url_map_traffic_director_path_partial
Success!
 The configuration is valid.
terraform_validate ./url_map_traffic_director_route
Success!
 The configuration is valid.
terraform_validate ./url_map_traffic_director_route_partial
Success!
 The configuration is valid.
terraform_validate ./vertex_ai_dataset
Success!
 The configuration is valid.
terraform_validate ./vertex_ai_featurestore
Success!
 The configuration is valid.
terraform_validate ./vertex_ai_managed_notebooks_runtime
Success!
 The configuration is valid.
terraform_validate ./vertex_ai_metadata_store
Success!
 The configuration is valid.
terraform_validate ./vertex_ai_user_managed_notebooks_instance
Success!
 The configuration is valid.
 ```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+1. Run `RUN_STAGE=init go test -v -timeout 0 -run //${SAMPLE_NAME}` to initialize
+    the sample to be tested.
+
+1. Run `RUN_STAGE=apply go test -v -timeout 0 -run //${SAMPLE_NAME}` to apply
+    the sample to be tested.
+
+1. Run `RUN_STAGE=verify go test -v -timeout 0 -run //${SAMPLE_NAME}` to verify
+    the sample has been applied.
+
+1. Run `RUN_STAGE=teardown go test -v -timeout 0 -run //${SAMPLE_NAME}` to destroy
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:72]: At each `RUN_STAGE` step, is this what users should expect to see in the output?

```
--- PASS: TestSamples (2.77s)
testing: warning: no tests to run
PASS
ok      github.com/terraform-google-modules/terraform-docs-samples/test/integration
```

If so, maybe we should mention this.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+1. Run `make docker_run` to start the testing Docker container in
+   interactive mode.
+
+1. Navigate to the test directory `cd test/integration`.
+
+1. Run `RUN_STAGE=init go test -v -timeout 0 -run //${SAMPLE_NAME}` to initialize
+    the sample to be tested.
+
+1. Run `RUN_STAGE=apply go test -v -timeout 0 -run //${SAMPLE_NAME}` to apply
+    the sample to be tested.
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:67]: Can users check to verify that the resources were created? Which project are they created in?

**Review by betsy-lichtenberg — CHANGES_REQUESTED** [review]

**Review by betsy-lichtenberg — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (betsy-lichtenberg) [reply CONTRIBUTING.md:36]: When I run in Cloud Shell, this step isn't necessary because Cloud Shell runs this command automatically.

**Comment by betsy-lichtenberg** [comment]
I approved the PR because I was able to get through the procedure successfully by using Cloud Shell.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./CONTRIBUTING.md:15:Integration tests are used to verify that the samples are actuatable 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ```

File: CONTRIBUTING.md (hunk 1 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:50]: I added this in just so that we have it documented if users are running locally.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./CONTRIBUTING.md:15:Integration tests are used to verify that the samples are actuatable 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ```

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: CONTRIBUTING.md (hunk 4 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:168]: @betsy-lichtenberg  that implies that not tests were run. Did you set a SAMPLE_NAME?
I also added sample outputs for each of these commands.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./CONTRIBUTING.md:89:TestSamples/1/storage_new_bucket command.go:185: 
./CONTRIBUTING.md:117:    TestSamples/1/storage_new_bucket 2022-10-28T02:24:59Z command.go:185: 
./CONTRIBUTING.md:157:    TestSamples/1/storage_new_bucket  command.go:185: 
./CONTRIBUTING.md:176:    TestSamples/1/storage_new_bucket command.go:185: 
./CONTRIBUTING.md:178:    TestSamples/1/storage_new_bucket command.go:185: 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ```

File: CONTRIBUTING.md (hunk 5 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:129]: Just pushed a change that should emit a logline for start and completion like `Test cloud_run_service_configuration completed in ci-tf-samples-2-7fc8 project`

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./CONTRIBUTING.md:89:TestSamples/1/storage_new_bucket command.go:185: 
./CONTRIBUTING.md:117:    TestSamples/1/storage_new_bucket 2022-10-28T02:24:59Z command.go:185: 
./CONTRIBUTING.md:157:    TestSamples/1/storage_new_bucket  command.go:185: 
./CONTRIBUTING.md:176:    TestSamples/1/storage_new_bucket command.go:185: 
./CONTRIBUTING.md:178:    TestSamples/1/storage_new_bucket command.go:185: 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ```

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+- [Docker Engine][docker-engine]
+- [Google Cloud SDK][google-cloud-sdk]
+- [make]
+
+## Linting and Formatting
+Files in the repository are linted or formatted to maintain a standard of quality and statically validated.
+You can run this check locally:
+
+```
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:17]: Can we add `cd terraform-docs-samples` here?

Comment (bharathkkb) [reply CONTRIBUTING.md:17]: Done

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+gcloud auth application-default login
+```
+
+With these settings in place, you can prepare a test project using Docker:
+
+```
+cd terraform-docs-samples
+make docker_test_prepare
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:54]: Can we add sample output here?

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

project_ids = [
  "ci-tf-samples-0-c5ab",
  "ci-tf-samples-1-f344",
  "ci-tf-samples-2-b5a7",
  "ci-tf-samples-3-6eda",
]
sa_key = <sensitive>
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+This document provides guidelines for contributing to samples.
+
+## Dependencies
+
+The following dependencies must be installed on the development system:
+
+- [Docker Engine][docker-engine]
+- [Google Cloud SDK][google-cloud-sdk]
+- [make]
+
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:12]: Can we mention that Cloud Shell already meets these requirements, so we recommend using Cloud Shell?

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+```
+
+### Interactive Execution
+
+Interactive execution is useful if you want to iteratively test a sample
+without destroying resources automatically.
+
+1. Run `make docker_run` to start the testing Docker container in
+   interactive mode.
+
```
Comment (betsy-lichtenberg) [inline CONTRIBUTING.md:104]: Here can we add:

The command prompt changes to `[root@8d7469507cf8 workspace]#`.

File: CONTRIBUTING.md (hunk 4 above)
Comment (betsy-lichtenberg) [reply CONTRIBUTING.md:169]: Looks good now.

File: CONTRIBUTING.md (hunk 5 above)
Comment (betsy-lichtenberg) [reply CONTRIBUTING.md:130]: Looks good

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: CONTRIBUTING.md (hunk 8 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:12]: done

File: CONTRIBUTING.md (hunk 9 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:127]: done

File: CONTRIBUTING.md (hunk 7 above)
Comment (bharathkkb) [reply CONTRIBUTING.md:56]: Added sample output, user need not know to know which project since the test framework handles the selection.

**Pull request merged**